lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/launch/JavaLaunchCommandBuilder.java

```diff
@@ -184,11 +185,17 @@ public JavaLaunchCommandBuilder setJavaPath(final String path) {
 
   public JavaLaunchCommandBuilder setClassPath(final String classPath) {
     this.classPath = classPath;
+    if (SystemUtils.IS_OS_WINDOWS) {
```

Comment:
Aren't we building the launch command for the target environment? SystemUtils.IS_OS_WINDOWS will give the OS of the current environment, which is not the same?

Reply:
The tricky part is that the java launch command in Windows and Linux is different. In Windows, the class path has to be -classpath 'com.abc.def'; In Linux, it is -classpath com.abd.def

Reply:
Compiling evnerionment and runtime enviroment are different . I will fix it.

Reply:
refactored code is uploaded. I am completely moving this to WindowsCommandBuilder and LinuxCommandBuilder, which makes more sense and does not need to make more conditions on OS.